### ISSUE TITLE
Bots can't operate properly in the higher pressure levels

### ISSUE THREAD

**Issue by itchyOwl** [Regalis11/Barotrauma#11439]
Found at least two inter-connected issues:

1. Bots don't know how much pressure the diving suit can handle, but with the recent changes they realize when they are not protected from the pressure, which I suspect can lead to getting stuck (in an objective loop?). So they should be able to handle situations where they can't find a suit that protects enough: yell about the pressure (-> add new dialogue), but still continue operating.
2. Bot don't prioritize between different suits -> can pick a regular suit over a combat diving suit, if it's the closest. Currently we have a system for telling them which suit they should use, but that doesn't take the pressure into account. Also it's only used for telling to prioritize PUCS over the other suits, and everything over exosuits (because they can't yet switch the battery cell).

**Comment by itchyOwl** [comment]
Addressed in Regalis11/Barotrauma-development#4118.

**Comment by Rokvach** [comment]
I was trying to test the diving suit prioritizing by spawning one PUCS suit next to regular suits in Typhon 2 in a biome where the outside pressure is too much for the regular diving suit and ordered a bot to prepare for expedition but the game crashed. 
```

Barotrauma Client crash report (generated on 04/25/2023 10:55:05)

Barotrauma seems to have crashed. Sorry for the inconvenience! 

6626DC3E580EC9988C7FF4945364D2BC

Game version 1.0.10.0 (ReleaseWindows, branch feature/bot-survival, revision ac3f8f4f47)
Graphics mode: 1280x720 (Windowed)
VSync OFF
Language: English
Selected content packages: Vanilla (BC446C6)
Level seed: Yasa94Seosu111
Loaded submarine: Typhon2 (C3A6D36219934C32AABF357E3181FCBB)
Selected screen: Barotrauma.GameScreen
SteamManager initialized

System info:
    Operating system: Microsoft Windows NT 10.0.19044.0 64 bit
    GPU name: NVIDIA GeForce GTX 980 Ti
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1.7777778}
    GPU status: Normal

Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Boolean FindSuitableContainer(Barotrauma.Character, Barotrauma.Item, System.Collections.Generic.List`1[Barotrauma.Item], Int32 ByRef, Barotrauma.Item ByRef)
Stack trace: 
   at Barotrauma.HumanAIController.FindSuitableContainer(Character character, Item containableItem, List`1 ignoredItems, Int32& itemIndex, Item& suitableContainer) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\HumanAIController.cs:line 932
   at Barotrauma.AIObjectiveCleanupItem.Act(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjectiveCleanupItem.cs:line 65
   at Barotrauma.AIObjective.TryComplete(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\Objectives\AIObjective.cs:line 179
   at Barotrauma.HumanAIController.Update(Single deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\AI\HumanAIController.cs:line 471
   at Barotrauma.Character.UpdateAll(Single deltaTime, Camera cam) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Characters\Character.cs:line 3037
   at Barotrauma.GameScreen.Update(Double deltaTime) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Screens\GameScreen.cs:line 186
   at Barotrauma.GameMain.Update(GameTime gameTime) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\GameMain.cs:line 881
   at Microsoft.Xna.Framework.Game.DoUpdate(GameTime gameTime) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 651
   at Microsoft.Xna.Framework.Game.Tick() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 500
   at Microsoft.Xna.Framework.SdlGamePlatform.RunLoop() in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\SDL\SDLGamePlatform.cs:line 93
   at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior) in <DEV>\Libraries\MonoGame.Framework\Src\MonoGame.Framework\Game.cs:line 397
   at Barotrauma.Program.Main(String[] args) in <DEV>\Barotrauma\BarotraumaClient\ClientSource\Program.cs:line 58
```

**Comment by itchyOwl** [comment]
Fixed in Regalis11/Barotrauma-development@c519ca1.

**Comment by Rokvach** [comment]
Tested, working correctly. Closing.
